File: 자료구조withPy/10866.py
import sys
from collections import deque
N=int(input())
Deque=deque([])

def isEmpty(tmp):
    if len(tmp)==0:return True
    else: return False

for _ in range(N):
    order=sys.stdin.readline().split()
    if order[0]=="push_front":Deque.appendleft(order[1])
    if order[0]=="push_back":Deque.append(order[1])
    if order[0]=="pop_front": #만약, 덱에 들어있는 정수가 없는 경우에는 -1을 출력한다.
        if(isEmpty(Deque)):print(-1)
        else:print(Deque.popleft())
    if order[0]=="pop_back": # 만약, 덱에 들어있는 정수가 없는 경우에는 -1을 출력한다.
        if(isEmpty(Deque)):print(-1)
        else:print(Deque.pop())
    if order[0]=="size":print(len(Deque))
    if order[0]=="empty": # 덱이 비어있으면 1을, 아니면 0을 출력한다.
        if(isEmpty(Deque)):print(1)
        else:print(0)
    if order[0]=="front": # 만약 덱에 들어있는 정수가 없는 경우에는 -1을 출력한다.
        if(isEmpty(Deque)):print(-1)
        else:print(Deque[0])
    if order[0]=="back": # 만약 덱에 들어있는 정수가 없는 경우에는 -1을 출력한다.
        if(isEmpty(Deque)):print(-1)
        else:print(Deque[-1])
    

File: 자료구조/10866.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});
let input = [];
let cnt = 0;
let N = 0;
let deque = [];

const isEmpty = () => {
  if (deque.length === 0) return true;
  else return false;
};

rl.on('line', (line) => {
  if (cnt === 0) N = parseInt(line.trim());
  else if (cnt <= N + 1) {
    input.push(line.trim().split(' '));
  }
  cnt++;
  if (cnt === N + 1) rl.close();
});
rl.on('close', () => {
  for (let i = 0; i < N; i++) {
    if (input[i][0] === 'push_back') deque.push(input[i][1]);
    else if (input[i][0] === 'push_front') deque.unshift(input[i][1]);
    else if (input[i][0] === 'pop_front') {
      if (isEmpty()) console.log(-1);
      else console.log(deque.shift());
    } else if (input[i][0] === 'pop_back') {
      if (isEmpty()) console.log(-1);
      else console.log(deque.pop());
    } else if (input[i][0] === 'size') console.log(deque.length);
    else if (input[i][0] === 'empty') {
      if (isEmpty()) console.log(1);
      else console.log(0);
    } else if (input[i][0] === 'front') {
      if (isEmpty()) console.log(-1);
      else console.log(deque[0]);
    } else if (input[i][0] === 'back') {
      if (isEmpty()) console.log(-1);
      else console.log(deque.at(-1));
    }
  }

  process.exit();
});
